fractal_server/app/routes/api/v2/task_collection.py

```diff
@@ -242,6 +253,85 @@ async def collect_tasks_pip(
     return state
 
 
+@router.post("/collect/custom/", status_code=201)
+async def collect_task_custom(
+    task_collect: TaskCollectCustomV2,
+    user: User = Depends(current_active_verified_user),
+    db: DBSyncSession = Depends(get_sync_db),
+):
+
+    settings = Inject(get_settings)
+
+    if task_collect.package_root is None:
+        if task_collect.package_name is None:
+            raise HTTPException(
+                status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+                detail=(
+                    "Must provide at least one of 'package_root' "
+                    "and 'package_name'."
```

Comment:
What about moving this check into a payload validator? This way we'll simplify the endpoint, and provide a fast response.